finagle-core/src/test/scala/com/twitter/finagle/InetResolverTest.scala

```diff
@@ -76,4 +78,36 @@ class InetResolverTest extends FunSuite {
     assert(statsReceiver.stat("lookup_ms")().size > 0)
   }
 
+  test("updates using custom future pool") {
+    class TestPool(latch: CountDownLatch) extends FuturePool {
+      private val delegate = FuturePool.unboundedPool
+      override def apply[T](f: => T): Future[T] = {
+        latch.countDown()
+        delegate(f)
+      }
+    }
+
+    val latch = new CountDownLatch(1) // DnsResolver will use the pool
+    val resolvePool = new TestPool(latch)
+
+    val dnsResolverWithPool = new DnsResolver(DefaultStatsReceiver, resolvePool)
+
+    def resolveLoopback(host: String): Future[Seq[InetAddress]] = {
+      if (host.equals("127.0.0.1")) dnsResolverWithPool(host)
+      else Future.exception(new UnknownHostException())
+    }
+
+    val pollInterval = Duration(100, TimeUnit.MILLISECONDS)
```

Comment:
I think the issue there is that `InetResolver` doesn't have a way to pass in a `Timer` and thus cannot use a `MockTimer`. It could be added as another constructor param, though that might not be worth it.